src/thunderbots/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -0,0 +1,396 @@
+
+/**
+ * This file contains the implementation of a theta star path planner
+ * which returns an optimal path that avoids obstacles
+ */
+
+
+ThetaStarPathPlanner::ThetaStarPathPlanner(Field field,
+                                           const std::vector<Obstacle> &obstacles)
+    : field_(field), obstacles_(obstacles)
+{
+    numRows = (int)(field_.totalLength() / SIZE_OF_GRID_CELL_IN_METERS);
+    numCols = (int)(field_.totalWidth() / SIZE_OF_GRID_CELL_IN_METERS);
+
+    unblocked_grid =
+        std::vector<std::vector<bool>>(numRows, std::vector<bool>(numCols, true));
+
+    for (unsigned row = 0; row < numRows; row++)
+    {
+        for (unsigned col = 0; col < numCols; col++)
+        {
+            Point p = convertCellToPoint(row, col);
+            for (unsigned index = 0; index < obstacles.size(); index++)
+            {
+                if (obstacles[index].getBoundaryPolygon().containsPoint(p))
+                {
+                    unblocked_grid[row][col] = false;
+                }
+            }
+        }
+    }
+}
+
+// A Utility Function to check whether given GridCell (row, col)
+// is a valid GridCell or not.
+bool ThetaStarPathPlanner::isValid(int row, int col)
+{
+    // Returns true if row number and column number
+    // is in range
+    return (row >= 0) && (row < numRows) && (col >= 0) && (col < numCols);
+}
+
+// A Utility Function to check whether the given GridCell is
+// blocked or not
+bool ThetaStarPathPlanner::isUnBlocked(int row, int col)
+{
+    return unblocked_grid[row][col];
+}
+
+// A Utility Function to check whether destination GridCell has
+// been reached or not
+bool ThetaStarPathPlanner::isDestination(int row, int col, CellCoordinate dest)
+{
+    return (row == dest.first && col == dest.second);
+}
+
+// A Utility Function to calculate the 'h' heuristics.
+double ThetaStarPathPlanner::calculateHValue(int row, int col, CellCoordinate dest)
+{
+    // Return using the distance formula
+    return ((double)sqrt((row - dest.first) * (row - dest.first) +
+                         (col - dest.second) * (col - dest.second)));
+}
+
+// A Utility Function to check for line of sight
+bool ThetaStarPathPlanner::lineOfSight(int curr_parent_i, int curr_parent_j,
+                                       CellCoordinate new_pair)
+{
+    Point nextPoint    = Point(new_pair.first, new_pair.second);
+    Point parentPoint  = Point(curr_parent_i, curr_parent_j);
+    Point pointToCheck = parentPoint;
+
+    Vector diff      = nextPoint - parentPoint;
+    Vector direction = diff.norm();
+    int dist         = (int)diff.len();
+    for (int i = 0; i < dist; i++)
+    {
+        pointToCheck = pointToCheck + direction;
+        if (!isUnBlocked((int)std::round(pointToCheck.x()),
+                         (int)std::round(pointToCheck.y())))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+// A Utility Function to trace the path from the source
+// to destination
```

Comment:
All these before function comments are great, could we move them to the javadoc comments? Just want to make sure we properly consolidate documentation to a consistent location.

Reply:
(if they're already in the javadoc comments, feel free to just remove them from here so we don't have duplicated documentation)

Reply:
Looks like there are other instances of this in other files, so could we apply this there as well?